nodejs/week2/src/backend/routes/meals.js

```diff
@@ -0,0 +1,45 @@
+const router = require('express').Router();
+const meals = require('../data/meals.json');
+
+router.get("/meals/:id", (req, res) => {
+    const mealId = Number(req.params.id);
+    const mealWithId = meals.filter((meal) => meal.id === mealId);
+    if (mealWithId.length > 0) {
+        res.send(mealWithId);
+    }
+    else {
+        res.status(404).send(`Can't find meal with id = ${mealId}`);
+    }
+});
+
+router.get("/meals", (req, res) => {
+
+    if (req.query.maxPrice) {
+        const maxPrice = Number(req.query.maxPrice);
+        const mealswithLesserPrice = meals.filter((meal) => meal.price < maxPrice);
+        res.send(mealswithLesserPrice);
+    }
+    else if (req.query.title) {
+        const mealTitle = (req.query.title).toLowerCase();
+        const mealsWithMatchingTitle = meals.filter((meal) => meal.title.toLowerCase().includes(mealTitle));
+        res.send(mealsWithMatchingTitle);
+    }
+    else if (req.query.createdAfter) {
+        const createdDate = new Date(req.query.createdAfter);
+        const mealsCreatedAfterDate = meals.filter((meal) => new Date(meal.createdAt) > createdDate);
+        res.send(mealsCreatedAfterDate);
+    }
+    else if (req.query.limit) {
+        const limit = Number(req.query.limit);
+        const mealsUptoLimit = meals.slice(0, limit);
+        res.send(mealsUptoLimit);
+    }
+    else if (req.originalUrl === '/meals') {
+        res.send(meals);// if original url returns all meals
+    }
```

Comment:
I think it is better to change condition, that sends array of meals, so it will not depend on route name.
(Otherwise you always should think about changing it, if route name is changed.)
Fx, you can check if req.query is an empty object.

```
if (Object.keys(req.query).length === 0 && req.query.constructor === Object) {
		response.json(meals);
	}
```
Maybe you can find a better way https://stackoverflow.com/questions/679915/how-do-i-test-for-an-empty-javascript-object